src/material/schematics/ng-update/migrations/legacy-components-v15/index.ts

```diff
@@ -6,12 +6,69 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Migration, TargetVersion} from '@angular/cdk/schematics';
 import * as ts from 'typescript';
+import * as postcss from 'postcss';
+import * as scss from 'postcss-scss';
+
+import {MIXINS} from './constants';
+
+import {Migration, ResolvedResource, TargetVersion, WorkspacePath} from '@angular/cdk/schematics';
 
 export class LegacyComponentsMigration extends Migration<null> {
   enabled = this.targetVersion === TargetVersion.V15;
 
+  override visitStylesheet(stylesheet: ResolvedResource): void {
+    let namespace: string | undefined = undefined;
+    const processor = new postcss.Processor([
+      {
+        postcssPlugin: 'legacy-components-v15-plugin',
+        AtRule: {
+          use: node => {
+            namespace = namespace ?? this._parseSassNamespace(node);
+          },
+          include: node => this._handleAtInclude(node, stylesheet.filePath, namespace),
+        },
+      },
+    ]);
+    processor.process(stylesheet.content, {syntax: scss}).sync();
+  }
+
+  /** Returns the namespace of the given at-rule if it is importing from @angular/material. */
+  private _parseSassNamespace(node: postcss.AtRule): string | undefined {
+    if (node.params.startsWith('@angular/material', 1)) {
+      return node.params.split(/\s/).pop();
```

Comment:
should it be `/\s+/` in case of multiple space?

Reply:
It doesn't matter for this case bc we only care about the last element, but I made the change anyways